### ISSUE TITLE
TcpEventBusBridge is incompatible with Service Proxy method of return value String

### ISSUE THREAD

**Issue by kovax** [vert-x3/vertx-tcp-eventbus-bridge#61]
### Context

I am migrating my open source framework from corba to vert.x using Service Proxies. The API is rather simple both method uses String as a return value. See the vert.x annotated interface here:
https://github.com/cristal-ise/cristal-ise/blob/develop/kernel/src/main/java/org/cristalise/kernel/entity/Item.java

The implementation is working when I use the eventBus directly, but when I try use the service through the TcpEventBusBridge I get this execption:
```
java.lang.ClassCastException: java.lang.String cannot be cast to io.vertx.core.json.JsonObject
	at io.vertx.ext.eventbus.bridge.tcp.impl.TcpEventBusBridgeImpl.lambda$doSendOrPub$3(TcpEventBusBridgeImpl.java:163)
	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:124)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:62)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:179)
	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
	at io.vertx.core.Promise.complete(Promise.java:66)
	at io.vertx.core.eventbus.impl.ReplyHandler.dispatch(ReplyHandler.java:97)
	at io.vertx.core.eventbus.impl.HandlerRegistration$InboundDeliveryContext.next(HandlerRegistration.java:163)
	at io.vertx.core.eventbus.impl.HandlerRegistration$InboundDeliveryContext.dispatch(HandlerRegistration.java:128)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:107)
	at io.vertx.core.eventbus.impl.HandlerRegistration.dispatch(HandlerRegistration.java:104)
	at io.vertx.core.eventbus.impl.ReplyHandler.doReceive(ReplyHandler.java:81)
	at io.vertx.core.eventbus.impl.HandlerRegistration.lambda$receive$0(HandlerRegistration.java:54)
```
The code (still under development) calling the TcpEventBusBridge can be found here:
https://github.com/cristal-ise/cristal-ise/blob/feature/Issue521_SwinGUI_VertxTcpEventbusBridge/kernel/src/main/java/org/cristalise/kernel/entity/proxy/ItemProxy.java#L265

The code for the TcpBridgeVerticle:
https://github.com/cristal-ise/cristal-ise/blob/feature/Issue521_SwinGUI_VertxTcpEventbusBridge/kernel/src/main/java/org/cristalise/kernel/process/TcpBridgeVerticle.java

### Do you have a reproducer?

Generated Service Proxy has these lines:

```java
    _vertx.eventBus().<String>request(_address, _json, _deliveryOptions, res -> {
      if (res.failed()) {
        returnHandler.handle(Future.failedFuture(res.cause()));
      } else {
        returnHandler.handle(Future.succeededFuture(res.result().body()));
      }
```

The `res` variable is of the type `AsyncResult<Message<String>>` which is clearly incompatible with the TcpEventBusBridge which uses `AsyncResult<Message<JsonObject>>` :

https://github.com/vert-x3/vertx-tcp-eventbus-bridge/blob/master/src/main/java/io/vertx/ext/eventbus/bridge/tcp/impl/TcpEventBusBridgeImpl.java#L147

**Comment by kovax** [comment]
Sorry, I submitted to issue prematurely, I will complete the original post shortly...

**Comment by vietj** [comment]
interesting thanks

**Comment by kovax** [comment]
@vietj the text is complete now :)

**Comment by kovax** [comment]
I can confirm, that TcpEventBusBridge works if I change the return type to JsonObject,

**Comment by pmlopes** [comment]
I've looked at the current clients:

* https://github.com/vert-x3/vertx-eventbus-bridge-clients/blob/master/python/vertx/eventbus.py#L34
* https://github.com/vert-x3/vertx-eventbus-bridge-clients/blob/master/nodejs/lib/tcp-vertx-eventbus.js#L210
* https://github.com/vert-x3/vertx-eventbus-bridge-clients/blob/master/javascript/vertx-eventbus.js#L261

There seems to be assumed that the `body` can be any valid JSON type while sending messages, which means we could already fix part of the issue at the bridge without breaking anything.

Receiving also seems to be assumed as any type on all these clients, so I'm confident that we can fix this bug internally without the need to introduce any breaking change.

**Comment by pmlopes** [comment]
Hi @kovax , I believe the PR above will fix this issue. Could you give it a try and see if it works for you too?

**Comment by kovax** [comment]
@pmlopes another solution (not bugfix) could be to generate a ServiceProxy specific TcpEventBusBridge, It seems to me that such a generator would actually eliminate most of my code, and the switch between using ServiceProxy and TCPBridge could be data/config driven.

**Comment by kovax** [comment]
> Hi @kovax , I believe the PR above will fix this issue. Could you give it a try and see if it works for you too?

Looks good to me. I am happy to try. Do you publish snapshot releases?

**Comment by vietj** [comment]
yes we do on s01.oss.sonatype repository

**Comment by kovax** [comment]
@pmlopes I confirm, that fix is worked using 4.2.0-SNAPSHOT
